code/client/src/components/AddressInput.jsx

```diff
@@ -0,0 +1,180 @@
+import { CheckOutlined, CloseOutlined, SearchOutlined } from '@ant-design/icons'
+import { Select, Space, Button, Tooltip } from 'antd'
+import Text from 'antd/lib/typography/Text'
+import React, { useCallback, useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import walletActions from '../state/modules/wallet/actions'
+import util, { useWindowDimensions } from '../util'
+
+/**
+ * Renders address input that provides type ahead search for any known addresses.
+ * Known addresses are addresses that have been entered by user for at least once.
+ */
+const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSelectOptions }) => {
+  const dispatch = useDispatch()
+
+  const wallets = useSelector(state => Object.keys(state.wallet.wallets).map((k) => state.wallet.wallets[k]))
+
+  const knownAddresses = useSelector(state =>
+    state.wallet.knownAddresses || {}
+  )
+
+  const network = useSelector(state => state.wallet.network)
+
+  const { isMobile } = useWindowDimensions()
+
+  const deleteKnownAddress = useCallback((address) => {
+    setAddressCallback({ value: '' })
+    dispatch(walletActions.deleteKnownAddress(address))
+  }, [dispatch])
+
+  const onSearchAddress = useCallback((address) => {
+    setAddressCallback({ value: address, label: address })
+  }, [setAddressCallback])
+
+  const walletsAddresses = wallets.map((wallet) => wallet.address)
+
+  /**
+   * Determines if the input wallet wallet is for the current wallet. Applicable for existing wallet management.
+   */
+  const notCurrentWallet = useCallback((inputWalletAddress) =>
+    !currentWallet || currentWallet?.address !== inputWalletAddress,
+  [currentWallet])
+
+  useEffect(() => {
+    const existingKnownAddresses = Object.keys(knownAddresses)
+      .map((address) => ({
+        address,
+        network: knownAddresses[address]?.network
+      }))
+
+    const walletsNotInKnownAddresses = wallets.filter((wallet) =>
+      !existingKnownAddresses.find((knownAddress) =>
+        knownAddress.address === wallet.address && knownAddress.network === wallet.network)
+    )
+
+    // Init the known address entries for existing wallets.
+    walletsNotInKnownAddresses.forEach((wallet) => {
+      dispatch(walletActions.setKnownAddress({
+        label: wallet.name,
+        address: wallet.address,
+        network: wallet.network,
+        creationTime: wallet.effectiveTime,
+        numUsed: 0
+      }))
+    })
+  }, [knownAddresses, wallets, dispatch])
```

Comment:
Turns out the issue is with this dependency `knownAddresses`. There might be some invalid wallets left in the user's state (e.g. my browser). If the invalid wallet has no valid `address`, it would become part of `walletsNotInKnownAddresses` and sent to dispatch for `setKnownAddress`. The reducer in `setKnownAddress` can't add that invalid wallet's address to known addresses, but would create a new object for `knownAddresses`, therefore trigger another `useEffect` here, hence causing an infinite loop

Reply:
I've made this effect run once when the component load.